src/status_im2/common/standard_authentication/standard_auth/authorize.cljs

```diff
@@ -12,33 +12,38 @@
-                                      :content (fn []
-                                                 [enter-password/view
-                                                  {:on-enter-password on-enter-password}])}]))})
-       (do
-         (reset-password)
-         (rf/dispatch [:show-bottom-sheet
-                       {:on-close on-close
-                        :theme    theme
-                        :shell?   blur?
-                        :content  (fn []
-                                    [enter-password/view
-                                     {:on-enter-password on-enter-password
-                                      :button-icon-left  auth-button-icon-left
-                                      :button-label      auth-button-label}])}]))))))
+  (let [password-login   (fn [{:keys [on-press-biometrics]}]
+                           (rf/dispatch [:show-bottom-sheet
+                                         {:on-close on-close
+                                          :theme    theme
+                                          :shell?   blur?
+                                          :content  (fn []
+                                                      [enter-password/view
+                                                       {:on-enter-password   on-auth-success
+                                                        :on-press-biometrics on-press-biometrics
+                                                        :button-icon-left    auth-button-icon-left
+                                                        :button-label        auth-button-label}])}]))
+        biometrics-login (fn [on-press-biometrics]
+                           (on-close)
+                           (rf/dispatch [:dismiss-keyboard])
+                           (biometric/authenticate
+                            {:reason     (i18n/label :t/biometric-auth-confirm-message)
+                             :on-success (fn [response]
+                                           (rf/dispatch [:standard-auth/on-biometric-success
+                                                         on-auth-success])
+                                           (log/info "response" response))
+                             :on-fail    (fn [error]
+                                           (log/error "Authentication Failed. Error:" error)
+                                           (when on-auth-fail (on-auth-fail error))
+                                           (password-login {:on-press-biometrics
+                                                            #(on-press-biometrics
+                                                              on-press-biometrics)}))}))]
+    (biometric/get-supported-type
+     (fn [biometric-type]
+       (if (and biometric-auth? biometric-type)
+         (biometrics-login biometrics-login)
+         (do
+           (reset-password)
+           (password-login {:on-press-biometrics nil})))))))
```

Comment:
It's a bit unconventional in Clojure to pass `nil` explicitly.

```clojure
;; Both return nil
(get {:on-press-biometrics nil} :on-press-biometrics) ; => nil
(get {} :on-press-biometrics) ; => nil
```

Reply:
sure, I can change that 👍